unity-ggjj/Assets/Scripts/TextDecoder/NarrativeScriptPlaylist.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class NarrativeScriptPlaylist : MonoBehaviour
+{
+    [field: Tooltip("List of narrative scripts to be played in order")]
+    [field: SerializeField] public List<NarrativeScript> NarrativeScripts { get; private set; }
+    
+    [field: Tooltip("List of narrative scripts to be used when taking penalties")]
+    [field: SerializeField] public List<NarrativeScript> FailureScripts { get; private set; }
+    
+    [field: Tooltip("A narrative script to be played on game over")]
+    [field: SerializeField] public NarrativeScript GameOverScript { get; private set; }
+    
+    private int _narrativeScriptIndex = -1;
+
+    /// <summary>
+    /// Initializes variables
+    /// </summary>
+    private void Awake()
+    {
+        NarrativeScripts.ForEach(script => script.Initialize());
+        FailureScripts.ForEach(script => script.Initialize());
+        GameOverScript.Initialize();
+    }
+
+    /// <summary>
+    /// Increments the narrative script index and returns the
+    /// corresponding script in the narrative scripts array
+    /// </summary>
+    /// <returns>The next narrative script in the array</returns>
+    public NarrativeScript GetNextNarrativeScript()
+    {
+        _narrativeScriptIndex++;
+        return _narrativeScriptIndex < NarrativeScripts.Count ? NarrativeScripts[_narrativeScriptIndex] : null;
+    }
+    
+    /// <summary>
+    /// Gets a random failure state from the list.
+    /// </summary>
+    /// <returns>Inky dialogue script containing a failure sub-story.</returns>
+    public NarrativeScript GetRandomFailureScript()
+    {
+        return FailureScripts.Count == 0 ? null : FailureScripts[Random.Range(0, FailureScripts.Count)];
```

Comment:
Nit: Similarly to the comment above, I would make the argument that, if this method is called and no failure scripts are available, we've entered an unexpected and unrecoverable state. (I was thinking about whether or not this could actually already be asserted during `Start()`, but if we're in a linear narrative, we might not need one as there's no way to fail.)

I could imagine something along those lines, if that makes sense to you too?
```suggestion
        if (FailureScripts.Count <= 0)
        {
            throw new NotSupportedException("This playlist contains no failure scripts");
        }
        return FailureScripts[Random.Range(0, FailureScripts.Count)];
```